kratos/python_scripts/multistage_analysis.py

```diff
@@ -0,0 +1,149 @@
+        '''This function executes (solves) the current stage.
+        Note that this call is equivalent to the traditional single-stage simulation run call.
+        '''
+
+        self.GetCurrentStage().Run()
+
+    def RunCurrentStagePostprocess(self):
+        '''This function executes the postprocessing of current stage.
+        Note that the stage postprocess deliberately involves operations only.
+        '''
+
+        if self.settings["stages"][self.GetCurrentStageName()].Has("stage_postprocess"):
+            if self.settings["stages"][self.GetCurrentStageName()]["stage_postprocess"].Has("operations"):
+                for operation in self.__GetOperations("stage_postprocess"):
+                    operation.Execute()
+
+    def GetNumberOfStages(self):
+        '''Returns the number of stages.'''
+
+        return len(self.__stages_map)
+
+    def GetCurrentStage(self):
+        '''Returns the current stage instance.'''
+
+        return self.__stages_map[self.__current_stage_name]
+
+    def GetCurrentStageName(self):
+        '''Returns the current stage name.'''
+        return self.__current_stage_name
+
+    def __GetExecutionList(self):
+        '''Creates and returns the execution list.
+        This method creates the execution list, either from a user-defined execution list or,
+        if this is not provided, from the stages declaration order.
+        '''
+
+        # Check if the execution list has been already created
+        if not hasattr(self, "__execution_list"):
+            # Default case in which the execution list is provided by the user
+            if self.settings.Has("execution_list"):
+                self.__execution_list = self.settings["execution_list"].GetStringArray()
+            # If not provided, create an auxiliary execution list from the stages declaration order
+            else:
+                KratosMultiphysics.Logger.PrintInfo("'execution_list' is not provided. Stages will be executed according to their declaration order.")
+                self.__execution_list = []
+                for stage_name in self.settings["stages"].keys():
+                    self.__execution_list.append(stage_name)
+
+        return self.__execution_list
+
+    def __GetModelers(self):
+        '''This method creates the modelers at the preprocess execution point.'''
+
+        execution_point_settings = self.settings["stages"][self.GetCurrentStageName()]["stage_preprocess"]
+
+        factory = KratosModelParametersFactory(self.model)
+        return factory.ConstructListOfItems(execution_point_settings["modelers"])
+
+    def __GetOperations(self, execution_point):
+        '''This method creates the operations at any execution point.'''
+
+        if execution_point not in ["stage_preprocess","stage_postprocess"]:
+            err_msg = f"Wrong execution point '{execution_point}'. Supported ones are 'stage_preprocess' and 'stage_postprocess'."
+            raise Exception(err_msg)
```

Comment:
Actually this is not right now, but considering that this is an still under development feature I put it in there just in case. Note that this error is checked twice per stage, so its performance affectation is indeed (almost) nothing.